client/src/main/kotlin/io/spine/examples/pingh/client/MentionsFlow.kt

```diff
@@ -0,0 +1,193 @@
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.pingh.client
+
+import com.google.protobuf.Duration
+import com.google.protobuf.util.Timestamps
+import io.spine.base.Time.currentTime
+import io.spine.examples.pingh.mentions.GitHubClientId
+import io.spine.examples.pingh.mentions.MentionId
+import io.spine.examples.pingh.mentions.MentionStatus
+import io.spine.examples.pingh.mentions.MentionView
+import io.spine.examples.pingh.mentions.UserMentions
+import io.spine.examples.pingh.mentions.UserMentionsId
+import io.spine.examples.pingh.mentions.buildBy
+import io.spine.examples.pingh.mentions.command.MarkMentionAsRead
+import io.spine.examples.pingh.mentions.command.SnoozeMention
+import io.spine.examples.pingh.mentions.command.UpdateMentionsFromGitHub
+import io.spine.examples.pingh.mentions.event.MentionRead
+import io.spine.examples.pingh.mentions.event.MentionSnoozed
+import io.spine.examples.pingh.mentions.event.MentionsUpdateFromGitHubCompleted
+import java.lang.Thread.sleep
+import kotlinx.coroutines.flow.MutableStateFlow
+
+/**
+ * The flow for managing the lifecycle of mentions.
+ *
+ * Enables updating mentions from GitHub, as well as reading and snoozing them.
+ *
+ * @param client enables interaction with the Pingh server.
+ * @param session the information about the current user session.
+ * @param settings the state of application settings.
+ */
+public class MentionsFlow internal constructor(
+    private val client: DesktopClient,
+    private val session: MutableStateFlow<UserSession?>,
+    private val settings: SettingsState
+) {
+    private companion object {
+        /**
+         * Delay before reading mentions so that the read-side on the server can be updated.
+         *
+         * Time is specified in milliseconds.
+         */
+        private const val delayBeforeReadingMentions = 100L
+    }
+
+    /**
+     * User mentions.
+     */
+    public val mentions: MutableStateFlow<MentionsList> = MutableStateFlow(findUserMentions())
+
+    /**
+     * Updates the user's mentions.
```

Comment:
I don't understand what it means, and I don't understand what it does. Neither from the description, nor from the method name. And this is a problem with such a documentation. It explains nothing.

Please review.